packages/front-end/components/DecisionCriteria/DecisionCriteriaTable.tsx

```diff
@@ -0,0 +1,229 @@
+          {decisionCriterias.map((criteria) => (
+            <tr
+              key={criteria.id}
+              className="hover-highlight"
+              onClick={() => handleRowClick(criteria)}
+              style={{ cursor: "pointer" }}
+            >
+              <td className="align-middle text-center">
+                {defaultCriteriaId === criteria.id && (
+                  <FaCheck color="var(--accent-9)" />
+                )}
+              </td>
+              <td className="align-middle">
+                <Flex direction="column" gap="1">
+                  <Box className="d-flex align-items-center gap-2">
+                    <Text weight="bold">{criteria.name}</Text>
+                    {isEditable(criteria) && (
+                      <Badge
+                        label="CUSTOM"
+                        color="violet"
+                        variant="soft"
+                        ml="2"
+                      />
+                    )}
+                  </Box>
+                  {criteria.description && (
+                    <Text color="gray" size="1">
+                      {criteria.description}
+                    </Text>
+                  )}
+                </Flex>
+              </td>
+              <td className="text-center align-middle">
+                {criteria.rules.length}
+              </td>
+              <td
+                className="text-right align-middle"
+                onClick={(e) => handleDropdownClick(e, criteria)}
+              >
+                <DropdownMenu
+                  trigger={
+                    <IconButton
+                      variant="ghost"
+                      color="gray"
+                      radius="full"
+                      size="3"
+                      highContrast
+                    >
+                      <BsThreeDotsVertical />
+                    </IconButton>
+                  }
+                  open={dropdownOpen === criteria.id}
+                  onOpenChange={(o) => {
+                    setDropdownOpen(o ? criteria.id : "");
+                  }}
+                >
+                  {isEditable(criteria) ? (
+                    <DropdownMenuItem
+                      onClick={() => {
+                        openCriteriaModal(criteria, false);
+                        setDropdownOpen("");
+                      }}
+                    >
+                      View/Edit
+                    </DropdownMenuItem>
+                  ) : (
+                    <DropdownMenuItem
+                      onClick={() => {
+                        openCriteriaModal(criteria, true);
+                        setDropdownOpen("");
+                      }}
+                    >
+                      View
+                    </DropdownMenuItem>
+                  )}
+
+                  <DropdownMenuItem
+                    onClick={() => {
+                      setDefaultCriteriaId(criteria.id);
+                      setDropdownOpen("");
```

Comment:
question: Do we need to manually handle the dropdown? I would expect simple open/pick flow to be supported without this kind of manual bookkepping.